### PR TITLE
Add targeted test for VectorDelta epsilon boundary

### PR DESCRIPTION
**Pull request by madmax983**
Added a targeted regression test to `src/core/version.rs` to ensure `VectorDelta` correctly handles floating point differences exactly equal to `VECTOR_EPSILON`. This eliminates a class of surviving mutants related to floating point comparison operators.

---
*PR created automatically by Jules for task [10805331307251116729](https://jules.google.com/task/10805331307251116729) started by @madmax983*

### CODE REVIEW

File: src/core/version.rs [hunk 1]
```diff
@@ -2685,4 +2685,41 @@ mod mutant_kill_tests {
...
+        );
+
+        // Case 3: Difference slightly smaller than EPSILON
+        // Should be considered equal
+        let v4 = vec![0.0f32 + VECTOR_EPSILON - 1e-9];
+        assert!(
+            VectorDelta::from_diff(&v1, &v4).is_none(),
+            "Difference < EPSILON should be considered equal"
+        );
+    }
```
[the remark this answers is not in the document]
Comment (madmax983) [reply src/core/version.rs:2727]: > medium
> 
> The test is well-structured and covers the epsilon boundary cases effectively. For improved readability and maintainability, consider extracting the magic number `1e-9` into a named constant and simplifying the vector initializations. This makes the intent of the small offset clearer and the code slightly more concise.
> 
> ```rust
>     #[test]
>     fn test_vector_delta_epsilon_boundary() {
>         // 🛡️ Sentinel Test: Verify exact epsilon handling in VectorDelta.
>         // This targets mutants that replace `<=` with `<` in floats_approx_equal.
>         // (a - b).abs() <= VECTOR_EPSILON
> 
>         // A small offset to test boundaries around VECTOR_EPSILON.
>         const SMALL_OFFSET: f32 = 1e-9;
> 
>         // Use 0.0 as base to ensure EPSILON is exactly representable and addition
>         // doesn't lose precision (which happens around 1.0 due to f32::EPSILON > VECTOR_EPSILON).
>         let v1 = vec![0.0f32];
> 
>         // Case 1: Difference exactly equal to EPSILON
>         // 0.0 + 1e-7. abs(diff) = 1e-7.
>         // Original (<=): Equal -> No Delta (None)
>         // Mutant (<): Not Equal -> Delta (Some)
>         let v2 = vec![VECTOR_EPSILON];
>         assert!(
>             VectorDelta::from_diff(&v1, &v2).is_none(),
>             "Difference of exactly EPSILON should be considered equal (no delta)"
>         );
> 
>         // Case 2: Difference slightly larger than EPSILON
>         // Should be detected as different
>         let v3 = vec![VECTOR_EPSILON + SMALL_OFFSET];
>         assert!(
>             VectorDelta::from_diff(&v1, &v3).is_some(),
>             "Difference > EPSILON should be detected"
>         );
> 
>         // Case 3: Difference slightly smaller than EPSILON
>         // Should be considered equal
>         let v4 = vec![VECTOR_EPSILON - SMALL_OFFSET];
>         assert!(
>             VectorDelta::from_diff(&v1, &v4).is_none(),
>             "Difference < EPSILON should be considered equal"
>         );
>     }
> ```
> 
> References

@jules
